internal/api/me_recently_viewed_docs.go

```diff
@@ -0,0 +1,129 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp-forge/hermes/internal/config"
+	"github.com/hashicorp-forge/hermes/pkg/models"
+	"github.com/hashicorp/go-hclog"
+	"gorm.io/gorm"
+)
+
+type recentlyViewedDoc struct {
+	ID      string `json:"id"`
+	IsDraft bool   `json:"isDraft"`
+}
+
+func MeRecentlyViewedDocsHandler(
+	cfg *config.Config,
+	l hclog.Logger,
+	db *gorm.DB,
+) http.Handler {
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		errResp := func(httpCode int, userErrMsg, logErrMsg string, err error) {
+			l.Error(logErrMsg,
+				"method", r.Method,
+				"path", r.URL.Path,
+				"error", err,
+			)
+			errJSON := fmt.Sprintf(`{"error": "%s"}`, userErrMsg)
+			http.Error(w, errJSON, httpCode)
+		}
+
+		// Authorize request.
+		userEmail := r.Context().Value("userEmail").(string)
+		if userEmail == "" {
+			errResp(
+				http.StatusUnauthorized,
+				"No authorization information for request",
+				"no user email found in request context",
+				nil,
+			)
+			return
+		}
+
+		switch r.Method {
+		case "GET":
+			// Find or create user.
+			u := models.User{
+				EmailAddress: userEmail,
+			}
+			if err := u.FirstOrCreate(db); err != nil {
+				errResp(
+					http.StatusInternalServerError,
+					"Error authorizing the request",
```

Comment:
nit: maybe "Error validating user information in request" would be more explicit of what error we want to show the user?

Reply:
This is more of an internal error where we've got an issue with the user data in our database vs. any issues with user info in the request (which is validated by the `auth` package), which is why I kind of favor using an internal error and this message instead of a 403.

Reply:
Sounds reasonable!